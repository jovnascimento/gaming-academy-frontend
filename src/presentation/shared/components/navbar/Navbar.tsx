import React, { useState } from "react";

import Box from "@mui/material/Box";
import AppBar from "@mui/material/AppBar";
import Toolbar from "@mui/material/Toolbar";
import { Button, Modal } from "@mui/material";
import SignIn from "../signin/SignIn";
import { Link } from "react-router-dom";

import Assets from "../../assets";
import SignUp from "../signup/SignUp";
import { useAppContext } from "../../contexts";

export const Navbar = () => {
  const appContext = useAppContext();

  const [loginModalOpen, setLoginModalOpen] = useState(false);
  const [signupModalOpen, setSignupModalOpen] = useState(false);

  const modalStyle = {
    position: "absolute" as "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    width: 600,
    bgcolor: "background.paper",
    boxShadow: 24,
    p: 4,
  };

  return (
    <>
      <Modal
        open={loginModalOpen}
        onClose={() => setLoginModalOpen(false)}
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
      >
        <Box sx={modalStyle}>
          <SignIn onFormFinish={() => setLoginModalOpen(false)} />
        </Box>
      </Modal>

      <Modal
        open={signupModalOpen}
        onClose={() => setSignupModalOpen(false)}
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
      >
        <Box sx={modalStyle}>
          <SignUp onFormFinish={() => setSignupModalOpen(false)} />
        </Box>
      </Modal>

      <AppBar position="static" color="secondary">
        <Toolbar>
          <Box
            style={{
              width: "100%",
              maxWidth: "1200px",
              display: "flex",
              margin: "0 auto",
            }}
          >
            <Box sx={{ flexGrow: 1 }}>
              <Button component={Link} to="/">
                <img
                  src={Assets.Logo}
                  alt="Gaming Academy"
                  style={{ height: 42 }}
                />
              </Button>
            </Box>
            <Box sx={{ flexGrow: 1 }} />

            {!appContext.token && (
              <>
                <Button
                  sx={{
                    borderWidth: 2,
                    borderStyle: "solid",
                    borderRadius: 50,
                    borderColor: "primary.main",
                    padding: "5px 20px",
                  }}
                  onClick={() => setSignupModalOpen(true)}
                >
                  Registrar
                </Button>
                <Button
                  sx={{ marginLeft: "15px" }}
                  onClick={() => setLoginModalOpen(true)}
                >
                  Login
                </Button>
              </>
            )}

            {appContext.token && (
              <>
                <Box sx={{ display: "flex", alignItems: "center" }}>
                  Bem-vindo, {appContext?.user?.name}
                </Box>

                {appContext.type === "instrutor" && (
                  <Button
                    sx={{ marginLeft: "15px" }}
                    component={Link} 
                    to="/add-course" 
                  >
                    Cadastrar Curso
                  </Button>
                )}

                <Button
                  sx={{ marginLeft: "15px" }}
                  onClick={() => {
                    appContext.setToken(null);
                    appContext.setUser(null);
                    appContext.setType(null);

                    localStorage.removeItem("token");
                    localStorage.removeItem("user");
                    localStorage.removeItem("type");
                  }}
                >
                  Logout
                </Button>
              </>
            )}
          </Box>
        </Toolbar>
      </AppBar>
    </>
  );
};
