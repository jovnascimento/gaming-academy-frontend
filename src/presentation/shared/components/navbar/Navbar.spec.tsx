import { renderWithProviders } from '../../../tests/utils'

import { Navbar } from './Navbar'

describe('presentation/Navbar', () => {
  it('should match snapshot', () => {
    const sut = <Navbar />
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const render = renderWithProviders(sut)
    expect(render).toMatchSnapshot()
  })
})
