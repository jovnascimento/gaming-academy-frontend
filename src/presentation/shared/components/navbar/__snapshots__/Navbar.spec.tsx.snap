// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`presentation/Navbar should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-ss5agw"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorSecondary MuiAppBar-positionStatic css-1wjib2q-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
          >
            <div
              class="MuiBox-root css-0"
              style="width: 100%; max-width: 1200px; display: flex; margin: 0px auto;"
            >
              <div
                class="MuiBox-root css-i9gxme"
              >
                <a
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1hn593o-MuiButtonBase-root-MuiButton-root"
                  href="/"
                  tabindex="0"
                >
                  <img
                    alt="Gaming Academy"
                    src="logo.png"
                    style="height: 42px;"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
              <div
                class="MuiBox-root css-i9gxme"
              />
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1f7ulx7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Registrar
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-lg1zh9-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Login
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-ss5agw"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorSecondary MuiAppBar-positionStatic css-1wjib2q-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <div
            class="MuiBox-root css-0"
            style="width: 100%; max-width: 1200px; display: flex; margin: 0px auto;"
          >
            <div
              class="MuiBox-root css-i9gxme"
            >
              <a
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1hn593o-MuiButtonBase-root-MuiButton-root"
                href="/"
                tabindex="0"
              >
                <img
                  alt="Gaming Academy"
                  src="logo.png"
                  style="height: 42px;"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </div>
            <div
              class="MuiBox-root css-i9gxme"
            />
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1f7ulx7-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Registrar
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-lg1zh9-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Login
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
